### DISCUSSION TITLE
[Frontend] Change the contents of the chosen language to change all saved contents

### DISCUSSION THREAD

**Discussion by ghost** [nasirkhan/laravel-starter#557] [Ideas]
Hello Nasir!

I was checking the demo and admin. I tried to create one category. But there is no variable to choose a language in which that category will be created. So it is saved without any language parameter.

Thus, this is a feature necessary, if the language is supposed to change by choosing in frontend a new language. Thereafter, it should pull only those contents, which were saved in that language.

This is not the case now.

**Comment by ghost** [comment]
Hello Nasir!

So sad that this was not considered as a feature request. There mmust be thousands of users, who can use your project, if this was a true multilanguage project.

At the moment, you only have a change of language in the front and backend of default language contents. This does not apply to new contents created in the database because the language parameter is not saved while creating it.

It does not take a lot of time for a professional like you to add one field of language_id of the new contenmt saved in the DB and filter the display based on this parameter $language_id to exclude all other contents having other languages.

By doing so, you would be making the same project with very little extra effort available for its use to thousands of users.

By shifting under discussion, you have forgotten it completely and developed other aspects and features, which is very nice. But this feature is necessary.

**Comment by nasirkhan** [comment]
Thank you for using Laravel Starter Template Project. I believe you are a long-time user of Laravel Starter. The feature you mentioned is not a must-have for most of the cases. But it is a good-to-have feature. For this reason, I did not want to spend time implementing this. 

There are several ways to implement translation in a project. It may require to have a list of all available or a module to add new languages. As per your preferences, post/category lists needed to be updated as well. Then the language list will not show the list of translations but rather will show a list from the language module. There may be other details of this multilingual/ translation feature.

This is a generic starter project that can be used in all types of projects, especially for custom application development.  But anyone may develop this feature and submit a PR, I would be very happy to accept that. I hope you will understand. 

If you are willing to sponsor me to develop this I would like to discuss in details about this feature with you.

**Comment by ghost** [comment]
My dear young friend Nasir!

`I believe you are a long-time user of Laravel Starter. `

I am not a user but have remained only as an observer. I had intended to use your project but after installing it I found you have developed this project on oneside with a lot of care and on the otherside delivered to this community a half cooked project. Thus, I have rejected use of your project before two years and am already using some other project for my non-commercial project.

Occassionally I update your project and visit, which still remains on my server. But I am not using it because it has some fundamental flaws and makes the entire project laughable.

After selecting a language, only the standard default language changes. But the dynamic contents that is created and saved in the database tables remains in a different languages. All of these contents will be pullout in the front end and will be shown.

I have never understood why you crteated such a gigantic defective project and that also being such a smart programmer injecting a huge defective feature of a half cooked multilanguage.

It would not downgrade your personality and competence, if you remove from description that this is a multilanguage project. If you offer a multilanguage, then offer it properly. But now you offered it as a joke, being a half cooked food.

If there is no variable saved in the database next to the categories and posts regarding languages, there is no change to display languages pulled out in that particular chosen language and display in the fronend.

So I have never understood why you designed this half cooked food. Thus, after understanding your style of prtesentation, I had to move forward and use some other packages without livewire. And I am happy with that and do not plan to change and do not want to use your project.

It is upto you to grasp my criticism and revise your project. I have mentioned only because I saw that you have put in this project a huge effort and have never understood why you presented this as a half cooked food and do not care to run even a small extra mile to make it complete.

`This is a generic starter project that can be used in all types of projects,`

Yes, this is a generic starter template. But under no circumstances this project can be classified under multilanguage without a field/colum attached to every dynamic record having a language_id attached to it.

I am now more than sure that you have not even understood what to do and how to make it multilanguage project. That is why you shifted (to your disadvantage, and not mine) under the discussion.

`Then the language list will not show the list of translations but rather will show a list from the language module. `

To have translations in the project is one thing. To habe dynamic contents displayed on the front end is a totally different thing.

You have only implemented translation part of the default contents. 

BUT YOU HAVE FAILED TO IMPLEMENT A TRUE MULTILANGUAGE FEATURE!

HENCE YOUR PROJECT IS LAUGHABLE ON USING IT BECAUSE THE FINAL RESULT WILL BE A MIX!

One can see this easily in your demo. After changing the language, you only see the translation but not the multilingual contents.

This is the most laughable part of the story. You don't care to understand how a multilanguage architecture works. You beleive that whatever you have done is multilanguage. THAT IS NOT MULTILANGUAGE BUT IT IS A HALF COOKED FOOD!

If you do not develop, I do not care. Even if you develop, I do not care as well because I am never ever going to be dependent on your project. But I am a well wisher of your work and project. I will always remain too, regardless of your laughable statements dreaming on multilanguage.
